### PR TITLE
feat: rewrite $head, $limit, and $tail to be equivalent to Python Polars

### PR DESCRIPTION
**Pull request by eitsupi**
I noticed that in Python, `polars.LazyFrame.head` calls the `slice` method internally.
https://github.com/pola-rs/polars/blob/740e740d9ce3678ea061d5cb4c2bc94892838383/py-polars/polars/lazyframe/frame.py#L4676-L4723

Here these functions are rewritten to match those in Python.
As a result, `<DataFrame>$head()` and `<DataFrame>$tail()` now support negative values.

### CODE REVIEW

File: R/lazyframe__lazy.R [hunk 1]
```diff
@@ -844,24 +844,31 @@ LazyFrame_sink_ndjson = function(
...
-  unwrap(.pr$LazyFrame$limit(self, n), "in $limit():")
-}
-
-#' @title Head of a LazyFrame
-#' @inherit DataFrame_head description params details
+#' Get the first `n` rows.
 #'
-#' @examples pl$LazyFrame(mtcars)$head(4)$collect()
+#' A shortcut for [`$slice(0, n)`][LazyFrame_slice].
+#' Conider using the [`$fetch()`][LazyFrame_fetch] method if you want to test your query.
```
Comment (etiennebacher) [inline R/lazyframe__lazy.R:850]:
```suggestion
#' Consider using the [`$fetch()`][LazyFrame_fetch] method if you want to test your query.
```

File: NEWS.md [hunk 2]
```diff
@@ -7,6 +7,8 @@
 -   In the when-then-otherwise expressions, the last `$otherwise()` is now optional,
     as in Python Polars. If `$otherwise()` is not specified, rows that don't respect
     the condition set in `$when()` will be filled with `null` (#836).
+-   `<DataFrame>$head()` and `<DataFrame>$tail()` methods now support negative
```
Comment (etiennebacher) [inline NEWS.md:10]: also the case for `LazyFrame`, right?

File: R/lazyframe__lazy.R [hunk 3]
```diff
@@ -844,24 +844,31 @@ LazyFrame_sink_ndjson = function(
...
+#' The [`$fetch()`][LazyFrame_fetch] operation will load the first `n` rows at
+#' the scan level, whereas `$head()` are applied at the end.
```
Comment (etiennebacher) [inline R/lazyframe__lazy.R:852]:
```suggestion
#' the scan level, whereas `$head()` is applied at the end.
```

**Review by etiennebacher — APPROVED** [review]
Thanks, LGTM, just a couple of typos.

I wonder why they keep both `limit()` and `head()` in the upstream polars if they are identical

File: NEWS.md (hunk 2 above)
Comment (eitsupi) [reply NEWS.md:10]: > also the case for `LazyFrame`, right?

No, LazyFrame does not know the total number of rows, so negative values are an error.

Comment (etiennebacher) [reply NEWS.md:10]: Right

**Review by etiennebacher — APPROVED** [review]

**Comment by eitsupi** [comment]
Thanks for your review.

> I wonder why they keep both `limit()` and `head()` in the upstream polars if they are identical

It is also strange that LazyFrame in Rust seems to have `limit` and `tail` but not `head`.

Perhaps the reason Python has `limit` as well as `head` is to match Rust.

**Pull request merged**